en/components/transaction-how-to.md

```diff
@@ -0,0 +1,69 @@
+
+With the accumulated state being a partial object, we can also use the service to check which cell has been edited and build UI around that.
+
+The `igxBaseTransactionService` has no undo stack so it does not provide undo/redo functionality.
+
+A detailed example of how you may use `igxBaseTransaction` to enable Row Editing is provided in the following topics:
+* [Grid Row Editing](grid/row_editing.md)
+* [Tree Grid Row Editing](treegrid/row_editing.md)
+* [Hierarchical Grid Row Editing](hierarchicalgrid/row_editing.md)
+
+### General information on igxTransactionService and igxHierarchicalTransactionService
+
+`igxTransactionService` and `igxHierarchicalTransactionService` are injectable middlewares, that implement the `TransactionService` interface. A component may use those to accumulate changes without affecting the underlying data. The provider exposes API to access, manipulate changes (undo and redo) and discard or commit all changes or a single change to the data.
+
+`igxTransactionService` and `igxHierarchicalTransactionService` work with both cell editing and row editing. The transaction for the cell edit is added when the cell exits edit mode, while row transaction is created, when the row exits edit mode. But in both cases the state of the grid edits consist of all updated, added and deleted rows and their last states. Those can later be inspected, manipulated and submitted at once. Changes are collected for individual cells or rows, depending on editing mode, and accumulated per data row/record.
+
+### Using igxTransactionService
+
+`igxTransactionService` extends `IgxBaseTransactionService`.
+
+You need to define the `igxTransactionService` as a provider for the igxGrid or for some of its parent components in order to have Batch Editing feature enabled.
+
+The `igxTransactionService` provides an undo stack so you may get advantage of the undo/redo functionality. The Undo stack is actually an array that contains array of transactions. When using the `igxTransactionService`, you may check the `canUndo` accessor in order to understand if there are any transactions in the Undo stack. If it does - you may use the `undo` method to remove the last transaction and `redo` to apply the last undone transaction.
+
+You may find a detailed example of how igxGrid with Batch Editing is implemented in the following topic:
+* [Grid Batch Editing](grid/batch_editing.md)
+
+### Using igxHierarchicalTransactionService
+
+`igxHierarchicalTransactionService` extends `IgxTransactionService`.
+
+The `igxHierarchicalTransactionService` is designed to handle the relations between parents and children in `igxHierarchicalGrid` and `igxTreeGrid` so that a new record will be added to the place you expect when adding an `ADD` transaction. And when you delete a parent record, its' children will be promoted to the higher level of hierarchy.
```

Comment:
`, its' children will be promoted to the higher level of hierarchy.`

the children could also be deleted. Change this to something like this:

`, its' children will be promoted to the higher level of hierarchy, or will be deleted with their parent.`